### PR TITLE
[Spark]Spark3 hive catalog junit

### PR DESCRIPTION
**Pull request by HuangFru**
fix create table in hive module
add getOrDefault in map in OverwriteBaseFiles
add mergeOnRead tests

### CODE REVIEW

File: spark/v3.1/spark/src/test/java/com/netease/arctic/spark/SparkTestContext.java [hunk 1]
```diff
@@ -450,4 +457,30 @@ public void assertHiveDesc(List<Object[]> rows, List<String> cols, List<String>
...
+      overwriteBaseFiles.commit();
+    } else if (table.isUnkeyedTable()) {
+      UnkeyedTable unkeyedTable = table.asUnkeyedTable();
+      OverwriteFiles overwriteFiles = unkeyedTable.newOverwrite();
+      Arrays.stream(dataFiles).forEach(overwriteFiles::addFile);
+      overwriteFiles.commit();
+    }else {
+      throw new IllegalStateException("Table is neither keyed nor unkeyed");
+    }
+  }
```
Comment (baiyangtx) [inline spark/v3.1/spark/src/test/java/com/netease/arctic/spark/SparkTestContext.java:484]: adapt hive write should in hive test context.   
all common method about hive should be in test context

File: spark/v3.1/spark/src/test/java/com/netease/arctic/spark/TestKeyedMergeOnRead.java [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class TestKeyedMergeOnRead extends SparkTestContext {
```
Comment (baiyangtx) [inline spark/v3.1/spark/src/test/java/com/netease/arctic/spark/TestKeyedMergeOnRead.java:22]: TestKeyedHiveTableMergeOnRead

File: spark/v3.1/spark/src/test/java/com/netease/arctic/spark/TestKeyedMergeOnRead.java [hunk 3]
```diff
@@ -0,0 +1,190 @@
...
+    testWrite(keyedTable, HiveLocationKind.INSTANT, Lists.newArrayList(
+        newRecord(keyedTable, 7, "aaa_hive", "2021-1-1"),
+        newRecord(keyedTable, 8, "bbb_hive", "2021-1-1"),
+        newRecord(keyedTable, 9, "ccc_hive", "2021-1-2"),
+        newRecord(keyedTable, 10, "ddd_hive", "2021-1-2")
+    ));
+
+    sql("select * from {0}.{1} order by id", database, table);
+    Assert.assertEquals(10, rows.size());
+    assertContainIdSet(rows, 0, 1,2,3,4,5,6,7,8,9,10);
```
Comment (baiyangtx) [inline spark/v3.1/spark/src/test/java/com/netease/arctic/spark/TestKeyedMergeOnRead.java:71]: why contain id=0 ?

File: spark/v3.1/spark/src/test/java/com/netease/arctic/spark/TestKeyedMergeOnRead.java [hunk 4]
```diff
@@ -0,0 +1,190 @@
...
+
+    List<Partition> partitions = hms.getClient().listPartitions(
+        database,
+        table,
+        (short) -1);
+    Assert.assertEquals(2, partitions.size());
+    sql("use spark_catalog");
+    sql("SET noUserSpecifiedNumPartition=false");
```
Comment (baiyangtx) [inline spark/v3.1/spark/src/test/java/com/netease/arctic/spark/TestKeyedMergeOnRead.java:79]: is this command necessary？

File: spark/v3.1/spark/src/test/java/com/netease/arctic/spark/TestKeyedMergeOnRead.java [hunk 5]
```diff
@@ -0,0 +1,190 @@
...
+    sql("select * from {0}.{1}", database, table);
+    Assert.assertEquals(4, rows.size());
+    assertContainIdSet(rows, 0, 7, 8, 9, 10);
```
Comment (baiyangtx) [inline spark/v3.1/spark/src/test/java/com/netease/arctic/spark/TestKeyedMergeOnRead.java:82]: why contain id=0?

File: spark/v3.1/spark/src/test/java/com/netease/arctic/spark/TestKeyedMergeOnRead.java [hunk 6]
```diff
@@ -0,0 +1,190 @@
...
+    sql("select * from {0}.{1} order by id", database, table);
+    Assert.assertEquals(10, rows.size());
+    assertContainIdSet(rows, 0, 1,2,3,4,5,6,7,8,9,10);
+    sql("use spark_catalog");
+    sql("SET noUserSpecifiedNumPartition=false");
+    sql("select * from {0}.{1}", database, table);
+    Assert.assertEquals(4, rows.size());
+    assertContainIdSet(rows, 0, 7, 8, 9, 10);
+  }
+
```
Comment (baiyangtx) [inline spark/v3.1/spark/src/test/java/com/netease/arctic/spark/TestKeyedMergeOnRead.java:189]: mor could add some position-delete tests

**Review by baiyangtx — CHANGES_REQUESTED** [review]

File: spark/v3.1/spark/src/test/java/com/netease/arctic/spark/TestKeyedMergeOnRead.java (hunk 5 above)
Comment (HuangFru) [reply spark/v3.1/spark/src/test/java/com/netease/arctic/spark/TestKeyedMergeOnRead.java:82]: that parameter is idIndex

File: spark/v3.1/spark/src/test/java/com/netease/arctic/spark/TestKeyedMergeOnRead.java (hunk 4 above)
Comment (HuangFru) [reply spark/v3.1/spark/src/test/java/com/netease/arctic/spark/TestKeyedMergeOnRead.java:79]: not necessary, i will delete it

**Review by baiyangtx — APPROVED** [review]

**Pull request merged**